cloudevents/working-drafts/xml-format.md

```diff
@@ -62,6 +62,11 @@ be expected from an XML document processing perspective.
 Where an event (or batch of events) is represented as a complete XML document,
 an XML document preamble SHOULD be included to ensure deterministic processing.
 
+XML comments are permitted anywhere within an XML representation of a
+CloudEvent or CloudEvent batch. Comments MUST be ignored during processing, except
+for the child element of a `<data>` element containing [XML data](#33-xml-data),
+where comments SHOULD be preserved.
```

Comment:
Perhaps this should be a MUST to ensure interop between SDKs?

Reply:
Been thinking about this one some more... and while I still believe that we should not be touching the "data" at all, and try to pass it along "as is" as best we can, perhaps it might be better to solve this at the spec level instead of at the format level. Then we don't need to say anything in this doc about data+comments at all.

If so, I then wonder about the other text related to comments since w/o the "except..." part of the sentence someone could interpret it as it being ok to remove comments from data. Perhaps rewording the "Comments MUST be ignored...." sentence as:
```
The presence of comments MUST NOT impact the CloudEvents metadata related processing.
```
This scopes it to just CE processing and not app-level processing.

Thoughts?

Reply:
Ah, but I said that about comments *with respect to XML*. I'm imagining a different text-based format that has very few options (e.g. no attributes) and where using comments for metadata is somewhat-idiomatic. That's the problem - if we move this to the spec, it needs to apply to all formats, including those we haven't thought of yet... although I guess we could always say "Unless the format says otherwise".

Anyway, let's get more input from other folks tomorrow. No head-messing was intended - just an added bonus ;)